src/linear_algebra/determinant.lean

```diff
@@ -49,6 +50,27 @@ local notation `ε` σ:max := ((sign σ : ℤ ) : R)
 definition det (M : matrix n n R) : R :=
 ∑ σ : perm n, ε σ * ∏ i, M (σ i) i
 
+/-- This is an alternate definition of `det` built up from primitives of `alternating_map`. -/
+def det_row_multilinear' : alternating_map R (n → R) R n :=
+((multilinear_map.mk_pi_algebra R n R).comp_linear_map (linear_map.proj)).alternatization
+
+/-- This is the more useful version of `det_row_multilinear'` that is defeq to `det`. -/
+@[simps]
+def det_row_multilinear : alternating_map R (n → R) R n :=
+(det_row_multilinear' : alternating_map R (n → R) R n).copy det $ begin
+  funext,
+  -- `multilinear_map.alternatization` and `det` use different spellings of multiplication by `ℤ`.
```

Comment:
We could probably change these two definitions to match and then I think this entire proof becomes `funext rfl`; but that will likely break every other proof about determinants, and I don't yet have a feel for which type of `ℤ`-multiplication is best.

Reply:
My vote is for `coe` + `mul` over `smul`, because `mul` has a bigger API and we have some nice results on how most maps can simplify to `coe : ℤ -> _`.

Why not "just" prove that `det_row_multilinear'` evaluates to the sum-of-permutations definition of `det`? (Perhaps even deleting the old definition of `det` altogether?) I don't see why the defeq-ness of `det` is something we want to preserve, especially since it's not even an especially nice definition, with a lot of `(-1) *` and `1 *` and `* 1` and `+ 0` in there.

Reply:
> My vote is for coe + mul over smul, because mul has a bigger API and we have some nice results on how most maps can simplify to coe : ℤ -> _.

My counter-argument is that `f (z • r)` rewrites to `z • f r` thanks to `linear_map.map_smul_of_tower`, but I think rewriting `f ((z : R) * r)`  to  `(z : R) * f r` would be a lot more work.

Thinking again, the `mul` / `coe` version isn't an option in `alternatization` anyway, since our codomain is not an algebra but a module.

> Why not "just" prove that det_row_multilinear' evaluates to the sum-of-permutations definition of det?

Well, that's thankfully what this `begin` / `end` block already does. I assume you're suggesting that I just sprinkle a few `rw det_apply`s in the existing proofs about `det` that I haven't touched in this PR to correct for the change in definition?